handlers/state-updates.ts

```diff
@@ -0,0 +1,122 @@
+import { octokit, DEVPOOL_OWNER_NAME, DEVPOOL_REPO_NAME } from "../helpers/constants";
+import { GitHubIssue, StateChanges } from "../types/github";
+
+export async function applyMetaChanges(
+  metaChanges: { title: boolean; body: boolean; labels: boolean },
+  devpoolIssue: GitHubIssue,
+  projectIssue: GitHubIssue,
+  isFork: boolean,
+  newLabels: string[]
+) {
+  const shouldUpdate = metaChanges.title || metaChanges.body || metaChanges.labels;
+
+  if (shouldUpdate) {
+    let newBody = devpoolIssue.body;
+
+    if (metaChanges.body && !isFork) {
+      newBody = projectIssue.html_url;
+    } else {
+      newBody = projectIssue.html_url.replace("https://", "https://www.");
+    }
+
+    try {
+      await octokit.rest.issues.update({
+        owner: DEVPOOL_OWNER_NAME,
+        repo: DEVPOOL_REPO_NAME,
+        issue_number: devpoolIssue.number,
+        title: metaChanges.title ? projectIssue.title : undefined,
+        body: metaChanges.body ? newBody : undefined,
+        labels: metaChanges.labels ? newLabels : undefined,
+      });
+    } catch (err) {
+      console.error(err);
+    }
+
+    console.log(`Updated metadata: ${devpoolIssue.html_url} - (${projectIssue.html_url})`, metaChanges);
+  }
+}
+
+export async function applyStateChanges(missingInPartners: boolean, projectIssue: GitHubIssue, devpoolIssue: GitHubIssue, taskHasPriceLabel: boolean) {
+  const stateChanges: StateChanges = {
+    // missing in the partners
+    forceMissing_Close: {
```

Comment:
Strange naming convention is this from GitHub? Otherwise you should fix

Reply:
It's to clearly signify the action which is being performed at a glance. It has been this way for some time, but i assume just drop the "_" is acceptable?